site/reference/fn/source/README.md

```diff
@@ -1,71 +1,64 @@
 ---
-title: "Source"
+title: "`source`"
 linkTitle: "source"
 type: docs
 description: >
-   Specify a directory as an input source package
+   Source resources from a local directory
 ---
 
 <!--mdtogo:Short
-    Specify a directory as an input source package
+    Source resources from a local directory
 -->
 
-Implements a [source function] by reading configuration and writing to STDOUT.
+`source` reads resources from a local directory and writes them in `functionSpec`
```

Comment:
Compare:

```
kpt fn source wordpress \
  | less
```

This is wrapped in ResourceList.

```
kpt fn source wordpress \
  | kpt fn eval --image gcr.io/kpt-fn/kubeval:v0.1 - \
  | less
```

This is multi-object YAML.


Is there a reason, all wire format as passed through pipes should not be the same:

```
kpt fn source wordpress \       
  | kpt fn eval --image gcr.io/kpt-fn/set-namespace:v0.1 - -- namespace=mywordpress \
  | kpt fn sink wordpress2
```

Need to also improve testing for ths.

Reply:
Related to whether we want flexibility of both wrapped and unwrapped with flags:

1. Source can emit output wrapped or unwrapped
2. Eval can accept input as wrapped or unwrapped
3. Eval can emit output as wrapped or unwrapped
4. Sink can accept input as wrapped or unwrapped

Reply:
For the OpenAPI scenario specifically: if we choose not to wrap the output in ResourceList, another option we'd discussed was wrapping the OpenAPI schema in its own KRM resource. 

I think either unwrapped or wrapped is fine, but I agree that it makes sense to have consistency across commands.

Reply:
> Is there a reason, all wire format as passed through pipes should not be the same:

I am not familiar with the reasoning. But when it comes to reading resources from a byte stream such as 'stdin`, all these command use same `kio.ByteReader` that supports resource in both (wrapped/unwrapped) format without specifying the format, so `eval` and `sink` supports both the formats. And when writing the output, they use `kio.ByteWriter` that writes in multi-yaml object format by default unless specified to wrap. For `fn source`, it always write the output in wrapped format.

So looking at the use case of creating your own pipeline using `source, eval, sink`, `source` needs to support `ResourceList` output format. `eval` needs to support `ResourceList` input as default (that is does) and it SHOULD output `ResourceList` by default (and that's where the confusion arises). and `sink` accepts `ResourceList` as input.
So only change of default output of `eval` to `ResourceList` will bring consistency and will bring it closer to the `pipeline` execution.

Now output to `multi-yaml` object has its benefits of being able to pipe to `kubectl` etc., so that can be supported with a flag.